python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py

```diff
@@ -13,6 +15,23 @@
 from mlcroissant._src.structure_graph.nodes.file_set import FileSet
 
 
+@functools.cache
+def _memoized_regex(pattern: str) -> re.Pattern:
+    regex = fnmatch.translate(pattern)
+    return re.compile(regex)
+
+
+def match(patterns: list[str] | None, path: pathlib.PurePath) -> bool:
```

Comment:
matches_path ?Maybe use a different name for this function? Match is a bit general maybe.

Reply:
Done.